src/PrestaShopBundle/Form/Admin/Improve/Payment/Preferences/PaymentModuleRestrictionsParentType.php

```diff
@@ -0,0 +1,78 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Form\Admin\Improve\Payment\Preferences;
+
+use PrestaShopBundle\Form\Admin\Type\TranslatorAwareType;
+use Symfony\Component\Translation\TranslatorInterface;
+
+/**
+ * Class PaymentModuleRestrictionsParentType defines is parent for all restriction forms in Payment Preferences
+ */
+class PaymentModuleRestrictionsParentType extends TranslatorAwareType
+{
+    /**
+     * @var array
+     */
+    protected $paymentModules;
+
+    /**
+     * @param TranslatorInterface $translator
+     * @param array $locales
+     * @param array $paymentModules
+     */
+    public function __construct(
+        TranslatorInterface $translator,
+        array $locales,
+        array $paymentModules
+    ) {
+        parent::__construct($translator, $locales);
+
+        $this->paymentModules = $this->sortPaymentModules($paymentModules);
+    }
+
+    /**
+     * Sort payment modules by display name.
+     *
+     * @param array $paymentModules
+     *
+     * @return array
+     */
+    protected function sortPaymentModules(array $paymentModules): array
+    {
+        $sortingBy = [];
+
+        foreach ($paymentModules as $key => $paymentModule) {
+            $sortingBy[$key] = $paymentModule->get('displayName');
+        }
+
+        array_multisort($sortingBy, SORT_ASC, $paymentModules);
```

Comment:
is it really a multidimensional array?
```suggestion
        array_multisort($sortingBy, $paymentModules);
```

Should be enough because SORT_ASC is the default value

Reply:
Well $paymentModules is multidimensional array. I tested without SORT_ASC and it seems to work just fine. So I am adding suggestion.